### PR TITLE
Separated ras tool tests for hw and mock platforms.

### PR DESCRIPTION
**Pull request by ibarraza**
Separated the valid test cases for ras to only run on MCP.
Separated the catastrophic and fatal error injection cases into mock only test cases.
Created a new test case for MCP hw only to run a valid test case but without catastropic and fatal error injection.

### CODE REVIEW

File: testing/ras/test_main_c.cpp [hunk 1]
```diff
@@ -434,6 +349,92 @@ class ras_c_mock_p : public ::testing::TestWithParam<std::string> {
...
+  strcpy(one, "--segment");
+  sprintf(two, "%d", platform_.devices[0].segment);
+  strcpy(three, "-B");
+  sprintf(four, "%d", platform_.devices[0].bus);
+  strcpy(five, "-D");
+  sprintf(six, "%d", platform_.devices[0].device);
+  strcpy(seven, "-F");
+  sprintf(eight, "%d", platform_.devices[0].function);
+  strcpy(nine, "-S");
+  if (platform_.devices[0].bus == 129) { strcpy(ten, "0x0"); }
```
Comment (tswhison) [inline testing/ras/test_main_c.cpp]: I thought we had a fix for this.  Maybe it hasn't made it to master yet?

File: testing/ras/test_main_c.cpp [hunk 2]
```diff
@@ -578,3 +579,91 @@ TEST_P(ras_c_mock_p, invalid_print_pwr_temp){
...
```
Comment (tswhison) [inline testing/ras/test_main_c.cpp]: Think we have a fix for this.

**Comment by tswhison** [comment]
Yeah, the fix is here in [885](https://github.com/OPAE/opae-sdk/pull/885).  It hasn't made its way to master yet.

**Comment by ibarraza** [comment]
@tswhison Ok. Can we get this in prior to #885 and update the ras tests in that PR? It still has a few fixes left in enum before that can be merged anyways.

**Pull request merged**